app/assets/javascripts/app/components/comment_form/comment_form.js

```diff
@@ -14,22 +15,33 @@
         comment: "=comment",
         discussion: "=discussion"
       },
+      controller: 'CommentFormCtrl as ctrl',
       templateUrl: 'components/comment_form/templates/comment_form.html',
-      link: function($scope, element, attrs) {
+    };
+  }
 
-        var clearBody = function() {
-          $scope.comment.body = '';
-        };
+  function CommentFormCtrl(DS, $scope) {
+    var ctrl = this;
 
-        var createComment = function(comment) {
-          $scope.discussion
-            .addComment(comment)
-            .then(clearBody);
-          $scope.toggle();
-        };
+    ctrl.clearBody = clearBody;
+    ctrl.updateComment = updateComment;
+    ctrl.createComment = createComment;
 
-        $scope.create = createComment;
-      }
-    };
+    function clearBody() {
+      $scope.comment.body = '';
+    }
+
+    function createComment(comment) {
+      $scope.discussion
+        .addComment(comment)
+        .then(clearBody);
+      $scope.toggle();
```

Comment:
Isn't this a string? How come you're calling it as a function? Am i missing something?